dashboard/app/controllers/lessons_controller.rb

```diff
@@ -44,14 +44,21 @@ def student_lesson_plan
     @lesson_data = @lesson.summarize_for_student_lesson_plan
   end
 
-  # GET /lessons/1/edit
+  # GET /s/csd1-2021/lessons/1/edit where 1 is the relative position of the lesson in the script
+  def edit_with_lesson_position
+    script = Script.get_from_cache(params[:script_id])
+    @lesson = script.lessons.find do |l|
+      l.has_lesson_plan && l.relative_position == params[:lesson_position].to_i
+    end
+
+    disallow_legacy_script_levels
```

Comment:
why only disallow these on one of the two edit paths?

Reply:
This line makes it happen for the other edit path https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/lessons_controller.rb#L5. We can't do the same thing for this edit path because we need to get `@lesson` before we can check to disallow_legacy_script_levels.